astropy/coordinates/sky_coordinate.py

```diff
@@ -1594,8 +1616,15 @@ class like ICRS, but not an instance ICRS() since the latter could have
                                  "new frame='{1}'.  Instead transform the coordinate."
                                  .format(coord_frame_cls.__name__, frame_cls.__name__))
 
+    # TODO: deprecate this in future
     if 'representation' in kwargs:
-        frame = frame_cls(representation=_get_repr_cls(kwargs['representation']))
+        repr_cls = _get_repr_cls(kwargs['representation'])
```

Comment:
And again. Also, I'd set `representation_type` as you have above. It does seem like you could make a mini-helper function `_normalize_representation_type(kwargs)`...